domain/repository/repository_service.go

```diff
@@ -56,5 +65,26 @@ func CreateService(urlToParse string, branch *string, configuration config.Confi
 		return nil, github.ErrUnsupportedGithubURL
 	}
 
-	return github.Create(urlSegments, branch, configuration)
+	url, err := url.Parse(configuration.GetAuthServiceURL())
+	if err != nil {
+		return nil, errors.Wrap(err, "auth service url not found")
+	}
+
+	authClient := client.New(goaclient.HTTPClientDoer(http.DefaultClient))
+	authClient.Host = url.Host
+	authClient.Scheme = url.Scheme
+	if goajwt.ContextJWT(*ctx) != nil {
+		authClient.SetJWTSigner(goasupport.NewForwardSigner(*ctx))
+	} else {
+		log.Logger().Info(ctx, nil, "no token in context")
+	}
+	tokenRetriever := tr.TokenRetriever{AuthClient: authClient, Context: ctx}
+	token, err := tokenRetriever.TokenForService(fmt.Sprintf("https://%s", githubHost))
```

Comment:
I think it would be better if we moved all of the authClient code into the token_retrieved.go code. It really isn't used anywhere else and only makes sense within the context of the token. 

I also don't think we need to have the TokenRetriever struct. We could probably just make the definition of the TokenForService something like: 
`func GetAccessToken(ctx *context.Context, u *url.URL, authURL string) (*string, error) {...}`

This way we have passed in everything we need for the function to work and also contained everything related to the token into one place.